core/src/bufr/BufrReader/BufrParser.cpp

```diff
@@ -92,6 +94,66 @@ namespace bufr {
         return exportedData;
     }
 
+    std::shared_ptr<DataContainer> BufrParser::parse(const eckit::mpi::Comm& comm)
+    {
+      auto msgsToParse = std::floor(file_.size() / comm.size());
+      size_t startOffset = comm.rank() * msgsToParse;
+
+      // Messages may not split evenly among tasks, so distribute the remaining messages
+      if (auto remander = file_.size() - comm.size() * msgsToParse)
+      {
+        if (comm.rank() < remander)
+        {
+          msgsToParse++;
+          startOffset += comm.rank();
+        }
+        else
+        {
+          startOffset += remander;
+        }
+      }
```

Comment:
Sorry to keep harping on this section of the code, but it seems overly complicated (again, could be that I don't understand it)

Why not just something like this:
```
auto msgsPerPE = std::floor(file_.size() / comm.size());
remainder = file_.size() % msgsPerPE;
startOffset = comm.rank()*msgsPerPE;
if (comm.rank() == comm.size()-1) {
  msgsPerPE += remainder;
}

Reply:
Ah, good point, I was being naive and thinking of messages not having subsets and that the overhead would be minimal. I retract my comment :-)